authors/apps/articles/urls.py

```diff
@@ -11,6 +11,7 @@
 app_name = 'articles'
 urlpatterns = [
     url(r'^', include(router.urls)),
+    url(r'^search/articles$', ArticleSearchList.as_view(), name="search"),
```

Comment:
@Mutuba I don't think we need an entirely new endpoint because it's the same resource, we are only retrieving a subset of the resource. Similarly, I don't think we need another view, the search functionality can be handled in the same ArticleViewSet

Reply:
Django REST has a filtering implementation that you can leverage
http://www.django-rest-framework.org/api-guide/filtering/

Reply:
After reading more on this, we can go with this implementation. Though is it possible to change the URL to 
`url(r'^articles/search$', ArticleSearchList.as_view(), name="search")` so it can match the retrieve/create articles url since they both reference the same resource?

Reply:
@WNjihia , When it is articles/search, it maps into the ArticleViewSet class instead of mapping into ArticleSearchList class!
So at the end it returns an error during GET request that the slug is not found, meaning it is treating the search params as a slug and not search args.

Reply:
url(r'^article/search$', ArticleSearchList.as_view(), name="search"
If I tweak, the articles to article, it works, so is it ok that way?

Reply:
Is it ok that way, @WNjihia

Reply:
I have a view the the url need to remain the same, because if the available option is to use [url(r'^article/search$',] and a user should search for multiple articles based on authors, titles and tags, then the url should be in plural. I tried, [url(r'^articles/search$', ArticleSearchList.as_view(), name="search")] and it was mapping into [ArticleViewSet ], so it cannot work for searching.

Reply:
@WNjihia ☝

Reply:
Would be best if we consult Tumbo on this. Will go ahead and merge this and will make changes based on feedback in a new PR